R/compute_long_short_returns.R

```diff
@@ -0,0 +1,27 @@
+#' Compute long-short portfolio returns
+#'
+#' Compute long-short portfolio returns
+#'
+#' @param portfolio_returns Some dataset...
+#'
+#' @return A tibble with long-short returns.
+#'
+#' @examples
+#' \donttest{
+#'   compute_long_short_returns(portfolio_returns)
+#' }
+#'
+#' @export
+compute_long_short_returns <- function(
+    portfolio_returns,
+    direction = "top_minus_bottom"
+) {
+  portfolio_returns |>
+    filter(portfolio %in% c(1, max(portfolio))) |>
+    mutate(portfolio = if_else(portfolio == 1, "bottom", "top")) |>
+    group_by(date) |>
+    summarize(long_short_return = ret_excess[portfolio == "top"] - ret_excess[portfolio == "bottom"],
```

Comment:
yields a warning if one of the portfolios is not populated for individual dates (e.g., because of clustering). Maybe pivot_wider before to return a silent NA.